{
    "source": "BDSA",
    "name": "BDSA-2019-3640",
    "title": "Pillow Vulnerable to Denial-of-Service (DoS) via Crafted Image Files",
    "description": "Pillow is vulnerable to denial-of-service (DoS) via crafted image files. An attacker able to supply a crafted invalid image file to a user of Pillow library may cause high memory consumption or extremely long processing time, resulting in a denial-of-service (DoS).",
    "technicalDescription": "This vulnerability was fixed by adding safety checks in `RawDecode.c ` and `PsdImagePlugin`, decompression bomb checks when parsing GIF and ICO formats, as well as a TIFF dimension type check.",
    "publishedDate": "2019-11-22T16:54:24.304Z",
    "updatedDate": "2020-02-25T16:24:01.112Z",
    "disclosureDate": "2019-10-07T00:00:00.000Z",
    "solution": "Fixed in [**6.2.0**](https://github.com/python-pillow/Pillow/releases/tag/6.2.0) by the following commits: \n* [Commit 1](https://github.com/python-pillow/Pillow/commit/87c43b66a5ffe0d21e04dce631a5fd4004e6021b)\n* [Commit 2](https://github.com/python-pillow/Pillow/commit/eed2bfc5b49782f33c5c73080b1d15abcc1e0c38)\n* [Commit 3](https://github.com/python-pillow/Pillow/commit/9a977b975cd871ef9a9128b72414c0de3a292591)\n* [Commit 4](https://github.com/python-pillow/Pillow/commit/ab52630d0644e42a75eb88b78b9a9d7438a6fbeb)",
    "severity": "LOW",
    "cvss2": {
        "baseScore": 1.9,
        "impactSubscore": 2.9,
        "exploitabilitySubscore": 3.4,
        "severity": "LOW",
        "accessVector": "LOCAL",
        "accessComplexity": "MEDIUM",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 1.4
        },
        "vector": "(AV:L/AC:M/Au:N/C:N/I:N/A:P/E:U/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 3.3,
        "impactSubscore": 1.4,
        "exploitabilitySubscore": 1.8,
        "severity": "LOW",
        "attackVector": "LOCAL",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "LOW",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "REQUIRED",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 2.9
        },
        "vector": "CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L/E:U/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "workaround": "",
    "vendorFixDate": "2019-09-30T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 2.9,
    "_meta": {
        "allow": [
            "GET"
        ],
        "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-3640",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro2.app.blackduck.com/api/cwes/CWE-400"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/CVE-2019-16865",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-3640/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugzilla.redhat.com/show_bug.cgi?id=1774066",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/commit/2f508d4467dcf82db01a79f5b11e9244a91921f2",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/commit/87c43b66a5ffe0d21e04dce631a5fd4004e6021b",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/commit/9a977b975cd871ef9a9128b72414c0de3a292591",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/commit/ab52630d0644e42a75eb88b78b9a9d7438a6fbeb",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/commit/eed2bfc5b49782f33c5c73080b1d15abcc1e0c38",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/issues/4123",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python-pillow/Pillow/releases/tag/6.2.0",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "default-remediation-status",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2019-3640/default-remediation-status"
            }
        ]
    }
}